java/src/main/java/com/amido/stacks/workloads/menu/service/v1/utility/MenuHelperService.java

```diff
@@ -0,0 +1,196 @@
+package com.amido.stacks.workloads.menu.service.v1.utility;
+
+import com.amido.stacks.workloads.menu.commands.CreateCategoryCommand;
+import com.amido.stacks.workloads.menu.commands.CreateItemCommand;
+import com.amido.stacks.workloads.menu.commands.DeleteCategoryCommand;
+import com.amido.stacks.workloads.menu.commands.DeleteItemCommand;
+import com.amido.stacks.workloads.menu.commands.UpdateCategoryCommand;
+import com.amido.stacks.workloads.menu.commands.UpdateItemCommand;
+import com.amido.stacks.workloads.menu.domain.Category;
+import com.amido.stacks.workloads.menu.domain.Item;
+import com.amido.stacks.workloads.menu.domain.Menu;
+import com.amido.stacks.workloads.menu.exception.CategoryAlreadyExistsException;
+import com.amido.stacks.workloads.menu.exception.CategoryDoesNotExistException;
+import com.amido.stacks.workloads.menu.exception.ItemAlreadyExistsException;
+import com.amido.stacks.workloads.menu.exception.ItemDoesNotExistsException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MenuHelperService {
+
+  public Optional<Category> findCategory(Menu menu, UUID categoryId) {
+    Optional<Category> existing = Optional.empty();
+    if (menu.getCategories() != null && !menu.getCategories().isEmpty()) {
+      existing =
+          menu.getCategories().stream()
+              .filter(c -> c.getId().equals(categoryId.toString()))
+              .findFirst();
+    }
+    return existing;
+  }
+
+  public Optional<Item> findItem(Category category, UUID itemId) {
+    Optional<Item> existing = Optional.empty();
+
+    if (category.getItems() != null && !category.getItems().isEmpty()) {
+      existing =
+          category.getItems().stream().filter(t -> t.getId().equals(itemId.toString())).findFirst();
+    }
+    return existing;
+  }
+
+  public Category getCategory(Menu menu, CreateItemCommand command) {
+    Optional<Category> existing = Optional.empty();
+
+    if (menu.getCategories() != null && !menu.getCategories().isEmpty()) {
+      existing =
+          menu.getCategories().stream()
+              .filter(c -> c.getId().equals(command.getCategoryId().toString()))
+              .findFirst();
+    }
+    return existing.orElseThrow(
+        () -> new CategoryDoesNotExistException(command, command.getCategoryId()));
+  }
+
+  public Category getCategory(Menu menu, DeleteCategoryCommand command) {
+    return findCategory(menu, command.getCategoryId())
+        .orElseThrow(() -> new CategoryDoesNotExistException(command, command.getCategoryId()));
+  }
+
+  public Category addItem(Category category, CreateItemCommand command) {
```

Comment:
I don't really like binding the Commands to this class
To me this class is for domain entities only
Better to use a mapper to convert from a command to a domain entity externally where this is called, and then pass just a domain entity in, otherwise you are mixing concerns

Reply:
fixed